sphinx_immaterial/content_tabs.py

```diff
@@ -0,0 +1,182 @@
+                ...content nodes
+    This allows for a default rendering in non-HTML outputs.
+    The ``MaterialTabSetHtmlTransform`` then transforms this container
+    into the HTML specific structure.
+    """
+
+    required_arguments = 1  # the tab label is the first argument
+    final_argument_whitespace = True
+    has_content = True
+    option_spec = {
+        "class": directives.class_option,
+    }
+
+    def run(self) -> List[nodes.Node]:
+        """Run the directive."""
+        self.assert_has_content()
+        if not is_md_tab_type(self.state_machine.node, "md-tab-set"):
+            LOGGER.warning(
+                "The parent of a 'md-tab-item' should be a 'md-tab-set'",
+                location=(self.env.docname, self.lineno),
+            )
+        tab_item = nodes.container(
+            "", is_div=True, type="md-tab-item", classes=["tabbed-block"]
+        )
+
+        # add tab label
+        textnodes, _ = self.state.inline_text(self.arguments[0], self.lineno)
+        tab_label = nodes.rubric(
+            self.arguments[0],
+            *textnodes,
+            classes=["tabbed-label"],
+        )
+        self.add_name(tab_label)
+        tab_item += tab_label
+
+        # add tab content
+        tab_content = nodes.container(
+            "",
+            is_div=True,
+            type="",
+            classes=["tabbed-block"] + self.options.get("class", []),
+        )
+        self.state.nested_parse(self.content, self.content_offset, tab_content)
+        tab_item += tab_content
+
+        return [tab_item]
+
+
+class content_tab_label(nodes.TextElement, nodes.General):
+    pass
+
+
+def visit_tab_label(self, node):
+    attributes = {"for": node["input_id"]}
+    self.body.append(self.starttag(node, "label", **attributes))
+
+
+def depart_tab_label(self, node):
+    self.body.append("</label>")
+
+
+def visit_tab_set(self: HTMLTranslator, node: content_tab_set):
+    # increment tab set counter
+    self.tab_set_count = getattr(self, "tab_set_count", 0) + 1
+
+    # configure tab set's div attributes
+    tab_set_identity = f"__tabbed_{self.tab_set_count}"
+    attributes = {"data-tabs": f"{self.tab_set_count}:{len(node.children)}"}
+    self.body.append(self.starttag(node, "div", **attributes))
+
+    # walkabout the children
+    tab_total = 0
+    tab_label_div = nodes.container("", is_div=True, classes=["tabbed-labels"])
+    tab_content_div = nodes.container("", is_div=True, classes=["tabbed-content"])
+    for tab_item in node.children:
+        try:
+            tab_label, tab_block = tab_item.children
+        except ValueError as exc:
+            raise ValueError(f"md-tab-item has no children:\n{repr(tab_item)}") from exc
+        tab_item_identity = tab_set_identity + f"_{tab_total + 1}"
+
+        # create: <input checked="checked" id="id" type="radio">
+        self.body.append(
+            "<input "
+            + ('checked="checked"' if not tab_total else "")
```

Comment:
It is also valid to just specify 'checked' --- for HTML boolean attributes it is not necessary to specify 'checked="checked"', though that is allowed:
https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes

In any case, though, I think you need a space afterwards, so that you don't end up with `<input checked="checked"type="radio" ...>`

Reply:
I can add the space which doesn't seem to be needed as the generated docs perform just the same (probably because the ending quotation mark).

As for the `check="checked"`, I'm just trying to mimic what the HTML looks like on upstream. If you find that the JS doesn't care about this attribute's value, then I can change it to a boolean value.

Notice also that the attribute never changes when selecting different tabs.

Reply:
Testing reveals that the quoted value is more resistant to missing spaces, but the boolean value works with the upstream JS just as well. I wish I could see the insider code so I won't have to retool this when that linked tabs feature goes public (~$900 away from that goal).

Reply:
now using boolean value